src/main/java/org/springframework/integration/dsl/context/IntegrationFlowContext.java

```diff
@@ -149,8 +149,11 @@ public void register(String flowId, IntegrationFlow integrationFlow, boolean aut
 	 */
 	public synchronized void remove(String flowId) {
 		if (this.registry.containsKey(flowId)) {
+			IntegrationFlow theFlow = this.registry.remove(flowId);
+			if (theFlow instanceof Lifecycle) {
+				((Lifecycle) theFlow).stop();
+			}
```

Comment:
Well, sounds like your suggestion to always start independently of the user preferences.
My idea is let to register flow and decide where to start it by the subsequent logic if you can't start it during registration.

I can utilize `isRunning()` logic here as well, but that should be covered in the target `SmartLifecycle` implementation.
For example the current `StandardIntegrationFlow.start()/stop()`  are idempotent and any double calls don't have effect.